extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetFlatPartitionedLayout.java

```diff
@@ -47,20 +59,26 @@ public String toString() {
     }
 
     @Override
-    public synchronized void findKeys(@NotNull final Consumer<ParquetTableLocationKey> locationKeyObserver) {
-        try (final DirectoryStream<Path> parquetFileStream =
-                Files.newDirectoryStream(tableRootDirectory.toPath(), ParquetFileHelper::fileNameMatches)) {
-            for (final Path parquetFilePath : parquetFileStream) {
-                ParquetTableLocationKey locationKey = cache.get(parquetFilePath);
-                if (locationKey == null) {
-                    locationKey = locationKey(parquetFilePath, readInstructions);
-                    if (!locationKey.verifyFileReader()) {
-                        continue;
+    public void findKeys(@NotNull final Consumer<ParquetTableLocationKey> locationKeyObserver) {
+        final SeekableChannelsProvider provider = SeekableChannelsProviderLoader.getInstance().fromServiceLoader(
+                tableRootDirectory, readInstructions.getSpecialInstructions());
+        try {
+            provider.applyToChildURIs(tableRootDirectory, uri -> {
+                if (!isNonHiddenParquetURI(uri)) {
+                    return;
+                }
+                synchronized (this) {
```

Comment:
I see this has moved from a coarse lock to a fine grained lock. I'm assuming this was on purpose?

Reply:
Note: `this` _is_ correct in the context of a lambda, but a bad refactor to anonymous class could miss this. In a nested context, I think it's best to be explicit, ie `ParquetFlatPartitionedLayout.this`.